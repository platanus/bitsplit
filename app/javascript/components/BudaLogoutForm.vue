<template>
  <div id="budaSignInForm">
    <center>
      <div class="w-full max-w-xs">
        <form @submit.prevent="handleSubmit">
          <div>
            <textField textvalue1="¿Estás seguro de querer desconectar tu cuenta Buda? Por favor ingresa tu contraseña para confirmar" />
          </div>
          <div>
            <inputLabel fieldName="Contraseña Bitsplit" fieldFor="password" />
            <passwordInput fieldId="password" fieldName="password" v-model="password"/>
          </div>
          <div>
            <submitButton :fieldDisabled="false" fieldPlaceholder="Confirmar" />
          </div>
        </form>
      </div>
    </center>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex'
import textInput from '../components/Input.vue'
import passwordInput from '../components/PasswordInput'
import submitButton from '../components/SubmitButton'
import inputLabel from '../components/InputLabel'
import textField from '../components/TextField'

export default {
  name: 'BudaLogoutForm',
  data() {
    return {
      routeName: 'BudaLogoutForm',
      api_key: '',
      api_secret: '',
      password: ''
    }
  },
  components: {
    textInput,
    passwordInput,
    submitButton,
    inputLabel,
    textField
  },
  computed: {
    ...mapState('user', ['currentUser'])
  },
  methods: {
    ...mapActions('user', ['budaSignIn']),
    handleSubmit(e) {
      const { api_key, api_secret, password } = this
      if ( password ) {
        const { email, authentication_token } = this.currentUser
        this.budaSignIn({ api_key, api_secret, password, email, authentication_token })
          .then(() => {
            console.log('success')
            this.$router.push('/home')
          })
          .catch(err => {
            console.error(err)
          })
      }
    }
  }
}
</script>

<style scoped>
#budaSignInForm {
  margin-top: 60px;
  margin-bottom: 60px;
}
</style>