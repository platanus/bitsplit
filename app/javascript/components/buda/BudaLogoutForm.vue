<template>
  <form @submit.prevent="handleSubmit">
    <div>
      <textField>
        ¿Estás seguro de querer desconectar tu cuenta Buda? Por favor ingresa tu
        contraseña para confirmar
      </textField>
    </div>
    <div>
      <inputLabel field-for="password">
        Contraseña Bitsplit
      </inputLabel>
      <passwordInput
        field-id="password"
        field-name="password"
        v-model="password"
      />
    </div>
    <div>
      <submitButton>Confirmar</submitButton>
    </div>
  </form>
</template>

<script>
import { mapActions, mapState } from 'vuex';
import passwordInput from '../PasswordInput';
import submitButton from '../SubmitButton';
import inputLabel from '../InputLabel';
import textField from '../TextField';

export default {
  name: 'BudaLogoutForm',
  data() {
    return {
      routeName: 'BudaLogoutForm',
      apiKey: '',
      apiSecret: '',
      password: '',
    };
  },
  props: {
    onBoardingScreen: {
      type: Boolean,
      default: false,
    },
  },
  components: {
    passwordInput,
    submitButton,
    inputLabel,
    textField,
  },
  computed: {
    ...mapState('user', ['currentUser']),
  },
  methods: {
    ...mapActions('user', ['budaSignOut']),
    ...mapActions('component', ['changeBudaComp']),
    handleSubmit() {
      const { apiKey, apiSecret, password } = this;
      if (password) {
        this.budaSignOut({
          api_key: apiKey,
          api_secret: apiSecret,
          password,
        })
          .then(() => {
            if (this.onBoardingScreen) {
              this.changeBudaComp('BudaForm');
            } else {
              this.$router.go();
            }
          })
          .catch(err => {
            console.error(err);
          });
      }
    },
  },
};
</script>
