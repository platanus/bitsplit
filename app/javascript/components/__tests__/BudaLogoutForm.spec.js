import { shallowMount } from '@vue/test-utils'
import BudaLogoutForm from '../BudaLogoutForm'

describe('BudaLogoutForm', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(BudaLogoutForm)
    expect(wrapper).toMatchSnapshot()
  })
})
