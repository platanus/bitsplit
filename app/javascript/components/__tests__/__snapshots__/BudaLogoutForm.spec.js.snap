// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudaLogoutForm matches snapshot 1`] = `
<form>
  <div>
    <textfield-stub>
      ¿Estás seguro de querer desconectar tu cuenta Buda? Por favor ingresa tu
      contraseña para confirmar
    </textfield-stub>
  </div>
  <div>
    <inputlabel-stub fieldfor="password">
      Contraseña Bitsplit
    </inputlabel-stub>
    <passwordinput-stub value="" fieldid="password" fieldname="password" fieldplaceholder="****************"></passwordinput-stub>
  </div>
  <div>
    <submitbutton-stub>Confirmar</submitbutton-stub>
  </div>
</form>
`;
